### PR TITLE
`Channel.lastMessageAt` as a computed property derived from the currently loaded messages

### PR DESCRIPTION
**Pull request by VelikovPetar**
### 🎯 Goal
We are currently calculating `Channel.lastMessageAt` as a computed property derived from the currently loaded messages. This can be wrong, if for example we have a deleted message as the last one in the channel, and we only fetch the last message from the channel. 

In this PR, this logic is reworked so that the `lastMessageAt` is initially populated from the `QueryChannels` call, and later updated by the relevant events (`message.new`, `notification.message_new`). This also aligns the logic with iOS. 

### 🛠 Implementation details

- Convert `lastMessageAt` from a computed property (derived from messages list) to a stored property on `Channel` and `ChannelData`
- Add `calculateNewLastMessageAt` utility function with proper filtering logic that skips shadowed messages, system messages (when configured), and thread replies not shown in channel
- Update `ChannelStateLogic`, `ChannelEventHandler`, and `EventHandlerSequential` to properly calculate and update `lastMessageAt` when messages change
- Update domain mapping and repository layers to persist `lastMessageAt`
- Add comprehensive unit tests for the new calculation logic

### 🎨 UI Changes

_Observe the **Other Group** channel_

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<video src="https://github.com/user-attachments/assets/c41184c1-a647-4011-8df1-e44b2563ae89" controls="controls" muted="muted" />
</td>
<td>
<video src="https://github.com/user-attachments/assets/2edd67d0-8863-4dd6-acd0-ac21b1062354" controls="controls" muted="muted" />
</td>
</tr>
</tbody>
</table>

### 🧪 Testing

- Unit tests added for `calculateNewLastMessageAt` function
- Existing tests updated to reflect the new property structure
- Manual testing: verify that channel list sorting by `lastMessageAt` works correctly

## Summary by CodeRabbit

* **Improvements**
  - Channels now maintain an explicit last message timestamp that updates whenever new messages are sent or channels are modified, ensuring accurate sorting and ordering in channel lists.

* **Tests**
  - Added comprehensive test coverage for last message timestamp calculations, including scenarios with system messages, thread replies, and timestamp edge cases.

### CODE REVIEW

File: stream-chat-android-state/src/main/java/io/getstream/chat/android/state/utils/internal/ChannelUtils.kt [hunk 1]
```diff
@@ -27,3 +30,35 @@ import io.getstream.chat.android.state.plugin.state.global.GlobalState
...
+    // Skip thread replies not shown in channel
+    if (message.parentId != null && !message.showInChannel) return currentLastMessageAt
+
+    val messageDate = message.createdLocallyAt ?: message.createdAt ?: return currentLastMessageAt
+
+    return if (currentLastMessageAt == null || messageDate.after(currentLastMessageAt)) {
+        messageDate
+    } else {
+        currentLastMessageAt
+    }
```
[the remark this answers is not in the document]
Comment (VelikovPetar) [reply stream-chat-android-state/src/main/java/io/getstream/chat/android/state/utils/internal/ChannelUtils.kt:63]: This is OK, because locally we always prioritise `createdLocallyAt`, to handle the case with optimistic updates when the current user sends a message. Also not that `createdLocallyAt` is only present in messages sent by the current user in the current session, and is never returned by the server.
